Sources/Prometheus/Histogram.swift

```diff
@@ -0,0 +1,174 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftPrometheus open source project
+//
+// Copyright (c) 2018-2023 SwiftPrometheus project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftPrometheus project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import CoreMetrics
+
+public protocol Bucketable: AdditiveArithmetic, Comparable {
+    var bucketRepresentation: String { get }
+}
+
+public typealias TimeHistogram = Histogram<Duration>
+public typealias ValueHistogram = Histogram<Double>
+
+public final class Histogram<Value: Bucketable>: @unchecked Sendable {
+    private let lock = NIOLock()
```

Comment:
Can we please use a `NIOLockedValueBox` here. We are really really encouraging everyone to not just put `@unchcked Sendable` on their types but rather stick everything into the locked value box since that only has condition `Sendable` conformance and we have come across a few places where people used a lock and incorrectly declared `@unchecked Sendable`.